dnf/plugins/repoquery/dnf-command-repoquery.c

```diff
@@ -115,17 +163,26 @@ dnf_command_repoquery_run (DnfCommand     *cmd,
 
   g_ptr_array_sort (pkgs, gptrarr_dnf_package_cmp);
 
-  // print packages without duplicated lines
   const char *prev_line = "";
   for (guint i = 0; i < pkgs->len; ++i)
     {
       DnfPackage *package = g_ptr_array_index (pkgs, i);
-      const char * line = dnf_package_get_nevra (package);
-      if (strcmp (line, prev_line) != 0)
+      if (opt_nevra || only_nevra)
+        {
+          const char * line = dnf_package_get_nevra (package);
+          // print nevras without duplicated lines
+          if (!only_nevra || strcmp (line, prev_line) != 0)
+            {
+              g_print ("%s\n", line);
+              prev_line = line;
+            }
+        }
+      if (opt_info)
         {
-          g_print ("%s\n", line);
-          prev_line = line;
+          print_package_info (package);
         }
+      if (!only_nevra)
+        g_print ("\n");
```

Comment:
... and here again, `only_nevra = !opt_info`, so you can put this line inside the `if (opt_info)` block above it.